finality-aleph/src/network.rs

```diff
@@ -233,24 +271,51 @@ enum MetaMessage {
     AuthenticationRequest(SessionId),
 }
 
+type MessageIndex = u64;
+
+#[derive(Clone, Encode, Decode)]
+struct DataMessage<D> {
+    session_id: SessionId,
+    index: MessageIndex,
+    data: D,
+    recipient: Recipient<NodeIndex>,
+}
+
+impl<D: Encode> Hash for DataMessage<D> {
```

Comment:
Maybe derive these (`Hash` and `PartialEq`) using the `derivative` crate, like in AlephBTF?

Reply:
Huh, I missed that, adding!

Reply:
Unfortunately, it doesn't work:( as D doesn't implement std::hash::Hash.

Reply:
Aw. :< Oh well.